kroxylicious-filters/kroxylicious-encryption/src/test/java/io/kroxylicious/filter/encryption/inband/InBandKeyManagerTest.java

```diff
@@ -587,6 +592,61 @@ void shouldEncryptRecordHeadersForMultipleRecords() throws ExecutionException, I
                 new TestingRecord(value2, new Header[]{ new RecordHeader("foo", new byte[]{ 10, 11, 12 }) })), decrypted);
     }
 
+    @Test
+    void decryptPreservesRecordOrder() {
+        var topic = "topic";
+        var partition = 1;
+
+        var kmsService = UnitTestingKmsService.newInstance();
+        InMemoryKms kms = kmsService.buildKms(new UnitTestingKmsService.Config());
+        var kekId1 = kms.generateKey();
+        var kekId2 = kms.generateKey();
+
+        var spyKms = Mockito.spy(kms);
```

Comment:
😱

Reply:
Hideous .. I'm hoping once we've got #774 etc we'd be able to rewrite this tests using mocks, and avoid the scary spying nonsense.

Reply:
Ohh that might be the forcing function I need to to actually merge #774